### PR TITLE
Download xml results and logs from on-device tests

### PR DESCRIPTION
**Pull request by oxve**
Android and raspi emit junit xml files and logs that Mobile Harness pulls from the devices
and uploads to the specified path on GCS.
Download those files and upload them as artifacts so the test report uploader will pick them up.

b/288331664
b/290997541

### CODE REVIEW

**Review by kaidokert — APPROVED** [review]
This looks ok to me.

**Review by isarkis — APPROVED** [review]

**Comment by oxve** [comment]
This change needs an internal change to be deployed before landing.

File: .github/actions/on_device_tests/action.yaml [hunk 1]
```diff
@@ -89,3 +92,22 @@ runs:
           --label author-${GITHUB_PR_HEAD_USER_LOGIN:-$GITHUB_COMMIT_AUTHOR_USERNAME} \
           --label author_id-${GITHUB_PR_HEAD_USER_ID:-$GITHUB_COMMIT_AUTHOR_EMAIL}
       shell: bash
+    - name: Download ${{ matrix.platform }} Test Results
+      if: env.TEST_TYPE == 'unit_test'
+      run: |
+        set -eux
+        gsutil -d cp "${GCS_RESULTS_PATH}" results.zip
```
Comment (isarkis) [inline .github/actions/on_device_tests/action.yaml:99]: I think with 'set -e' option bash will exit if line 99 fails.  You probably need to set -e after you check gsutil return status.

**Review by isarkis — CHANGES_REQUESTED** [review]

File: .github/actions/on_device_tests/action.yaml (hunk 1 above)
Comment (oxve) [reply .github/actions/on_device_tests/action.yaml:99]: Removing that check altogether instead. It doesn't add anything useful over the regular step failed status.

File: .github/actions/on_device_tests/action.yaml [hunk 2]
```diff
@@ -89,3 +92,22 @@ runs:
...
+        if [[ $? -ne 0 ]]; then
```
Comment (isarkis) [inline .github/actions/on_device_tests/action.yaml:100]: This check will be removed?

Comment (oxve) [reply .github/actions/on_device_tests/action.yaml]: I pushed the branch to my fork instead of here. Should be updated now.

**Review by isarkis — APPROVED** [review]

**Comment by oxve** [comment]
I pushed a new commit that accommodates the internal change that also uploads the logs files to GCS.

File: .github/actions/on_device_tests/action.yaml [hunk 3]
```diff
@@ -79,11 +78,11 @@ runs:
           ${DIMENSION:+"--dimension" "$DIMENSION"} \
           ${USE_SHARDING:+"--unittest_shard_index" "${{ matrix.shard }}"} \
           ${ON_DEVICE_TEST_ATTEMPTS:+"--test_attempts" "$ON_DEVICE_TEST_ATTEMPTS"} \
-          --archive_path ${GCS_PATH}/artifacts.tar \
-          --gcs_xml_result_path=${GCS_RESULTS}
+          --archive_path "${GCS_PATH}/artifacts.tar" \
+          --gcs_xml_result_path "${GCS_PATH}/results/${GITHUB_RUN_ID}/"
```
Comment (kaidokert) [inline .github/actions/on_device_tests/action.yaml:82]: The arg in python script was renamed to `gcs_rseult_path` - are those intentionally different ?

File: tools/on_device_tests_gateway.proto [hunk 4]
```diff
@@ -52,6 +52,7 @@ message OnDeviceTestsCommand {
   string start_timeout = 20;
   string test_timeout = 21;
   string builder_url = 22;
+  string gcs_xml_result_path = 23;
```
Comment (kaidokert) [inline tools/on_device_tests_gateway.proto:55]: Still different naming here and in python script 😄

File: .github/actions/on_device_tests/action.yaml (hunk 3 above)
Comment (oxve) [reply .github/actions/on_device_tests/action.yaml]: Done

File: tools/on_device_tests_gateway.proto (hunk 4 above)
Comment (oxve) [reply tools/on_device_tests_gateway.proto]: Ugh, done

**Comment by kaidokert** [comment]
Pretty sure this PR is stale by now !

**Pull request closed without merging**